lib/tool_shed/metadata/repository_metadata_manager.py

```diff
@@ -1107,3 +1066,39 @@ def _get_changeset_revisions_that_contain_tools(app: "ToolShedApp", repo, reposi
                 if metadata.get("tools", None):
                     changeset_revisions_that_contain_tools.append(changeset_revision)
     return changeset_revisions_that_contain_tools
+
+
+def get_user_by_username(session, username, user_model):
```

Comment:
Why is the user_model here the third and below it always the second argument?
Does the model really need to be passed in? Can we not assume its the user model by the function name?

Reply:
Hmm.. I passed all those models to these functions because in a different module directly importing them from `tool_shed.webapp.model` raised an error due to a circular dependency. However, those were modules in `util`, but this is in `managers`, so there should be no error from an import. Let me clean this up a bit (I hate passing those models).

Reply:
I am not reusing the function from `galaxy.managers.users` because that one calls `first()`, whereas this one calls `one()`, so the behavior is slightly different. Most likely the difference is inconsequential, but with the tool shed I cannot be sure. And the calling code would have to be slightly adjusted. Given the scale of these changes, I'm trying to minimize my footprint here.